storage/azure/src/main/java/io/aiven/kafka/tieredstorage/storage/azure/MetricCollector.java

```diff
@@ -0,0 +1,108 @@
+
+import java.util.List;
+
+import org.apache.kafka.common.MetricNameTemplate;
+import org.apache.kafka.common.metrics.JmxReporter;
+import org.apache.kafka.common.metrics.KafkaMetricsContext;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.CumulativeCount;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.utils.Time;
+
+import io.aiven.kafka.tieredstorage.metrics.SensorProvider;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelineNextSyncPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+public class MetricCollector {
+    private final org.apache.kafka.common.metrics.Metrics metrics;
+
+    private static final String METRIC_GROUP = "azure-blob-metrics";
+
+    MetricCollector() {
+        final JmxReporter reporter = new JmxReporter();
+
+        metrics = new org.apache.kafka.common.metrics.Metrics(
+            new MetricConfig(), List.of(reporter), Time.SYSTEM,
+            new KafkaMetricsContext("aiven.kafka.server.tieredstorage.azure")
+        );
+    }
+
+    MetricsPolicy policy() {
+        return new MetricsPolicy(metrics);
+    }
+
+    static class MetricsPolicy implements HttpPipelinePolicy {
+
+
+        private final Sensor deleteObjectRequests;
+        private final Sensor uploadObjectRequests;
+        private final Sensor getObjectRequests;
+
+        private final Metrics metrics;
+
+        MetricsPolicy(final Metrics metrics) {
+            this.metrics = metrics;
+            this.deleteObjectRequests = createSensor("object-delete");
+            this.uploadObjectRequests = createSensor("object-upload");
+            this.getObjectRequests = createSensor("object-get");
+        }
+
+        private Sensor createSensor(final String name) {
+            return new SensorProvider(metrics, name)
+                .with(new MetricNameTemplate(name + "-rate", METRIC_GROUP, ""), new Rate())
+                .with(new MetricNameTemplate(name + "-total", METRIC_GROUP, ""), new CumulativeCount())
+                .get();
+        }
+
+        @Override
+        public Mono<HttpResponse> process(final HttpPipelineCallContext context, final HttpPipelineNextPolicy next) {
+            processMetrics(context);
+            return next.process();
+        }
+
+        @Override
+        public HttpResponse processSync(final HttpPipelineCallContext context, final HttpPipelineNextSyncPolicy next) {
+            processMetrics(context);
+            return next.processSync();
+        }
+
+        void processMetrics(final HttpPipelineCallContext context) {
+            switch (context.getHttpRequest().getHttpMethod().name()) {
```

Comment:
All three should have a path like `/<container>/<blob>` or (`/<account>/<container>/<blob>` on Azurite). There's difference in verbs and also in query.

There is no price difference between `PutBlob`, `PutBlock`, and `PutBlockList`. All three have the same verb (`PUT`) and the same path (/account/container/blob), but differ in query params only. We probably don't need to distinct them for now.

I think it's probably not needed to prepare for some hypothetical future adding of more operation which need distinction by the path and query, but it makes sense to leave a comment about this, maybe in `AzureBlobStorage.java` to be more visible.